src/polyglot-notebooks-vscode-common/src/lineReader.ts

```diff
@@ -15,12 +15,12 @@ export class LineReader {
 
         let i = this.current.indexOf('\n');
         while (i >= 0) {
-            let line = this.current.substr(0, i);
+            let line = this.current.substring(0, i);
```

Comment:
`substr(start, length?)` is deprecated. However all use of it translated to `substring(start, end?)` directly.